core/src/main/scala/org/bitcoins/core/wallet/utxo/InputSigningInfo.scala

```diff
@@ -15,9 +16,24 @@ import org.bitcoins.crypto.Sign
   */
 sealed trait InputSigningInfo[+InputType <: InputInfo] {
   def inputInfo: InputType
+  def prevTransaction: Transaction
   def hashType: HashType
   def signers: Vector[Sign]
 
+  // If using EmptyTransaction we are testing or dummy signing
+  require(
+    prevTransaction == EmptyTransaction || outPoint.txId == prevTransaction.txId,
```

Comment:
is there a way the `prevTransaction == EmptyTransaction` invariant can be used to exploit the signature serialization scheme?

Reply:
I added ```
    require(
      signingInfo.prevTransaction != EmptyTransaction,
      "prevTransaction can only be an EmptyTransaction when dummy signing")
```
to `serializeForSignature`, this way an error will be thrown if we are doing a real signature, not a dummy signature or one that expects an error hash.